data_structures/heap/min_heap.py

```diff
@@ -40,10 +40,10 @@ def __getitem__(self, key):
         return self.get_value(key)
 
     def get_parent_idx(self, idx):
-        if len(self.heap_dict) % 2 == 0:
-            return idx // 2
+        if (len(self.heap_dict) - 1) % 2 == 0:
```

Comment:
if len(self.heap-dict) % 2:

Reply:
I was running this min-heap implementation on a Dijkstra's shortest distance problem in a separate module with a lot of vertex data and was getting sometimes correct but other times wrong results until I tweaked the parent index function and some other calculations. Not sure how I'd put that in a simple doctest here. This seems to run fine with small inputs, so I'm going to leave it be. Thanks for taking the time to review!